plugin/hosts/hosts_test.go

```diff
@@ -6,87 +6,158 @@ import (
 	"testing"
 
 	"github.com/coredns/coredns/plugin/pkg/dnstest"
+	"github.com/coredns/coredns/plugin/pkg/fall"
 	"github.com/coredns/coredns/plugin/test"
 
 	"github.com/miekg/dns"
 )
 
+// HostsCase extends test.Case to use Fallthrough flag.
+type HostsCase struct {
+	Fallthrough bool
+	Case        test.Case
+}
+
 func TestLookupA(t *testing.T) {
-	h := Hosts{
-		Next: test.ErrorHandler(),
-		Hostsfile: &Hostsfile{
-			Origins: []string{"."},
-			hmap:    newMap(),
-			inline:  newMap(),
-			options: newOptions(),
-		},
-	}
-	h.hmap = h.parse(strings.NewReader(hostsExample))
+	for i, tc := range hostsTestCases {
```

Comment:
adding the test is good, but why did you amend an existing test? Better to just have a small contained test with only 1 use case: one that fails now and is fixed with this PR

Reply:
the Case structure doesn't have any optional fields, so I need to extend it to use `Fallthrough` flag in `func TestLookupA`, I could not add `Fallthrough` only for the last test case.
So... should I add a new function like `func TestFallthrough` which checks only one case?
I don't have an idea which one is better, I'm going to follow your advice.
Thanks!